cpp/src/arrow/json/chunker.h

```diff
@@ -41,17 +42,23 @@ class ARROW_EXPORT Chunker {
 
   /// \brief Carve up a chunk in a block of data to contain only whole objects
   /// \param[in] block json data to be chunked
-  /// \param[out] chunked subrange of block containing whole json objects
-  virtual Status Process(util::string_view block, util::string_view* chunked) = 0;
+  /// \param[out] whole subrange of block containing whole json objects
+  /// \param[out] partial subrange of block a partial json object
+  virtual Status Process(const std::shared_ptr<Buffer>& block,
+                         std::shared_ptr<Buffer>* whole,
+                         std::shared_ptr<Buffer>* partial) = 0;
 
   /// \brief Carve the completion of a partial object out of a block
   /// \param[in] partial incomplete json object
   /// \param[in] block json data
-  /// \param[out] completion subrange of block contining the completion of partial
-  virtual Status Process(util::string_view partial, util::string_view block,
-                         util::string_view* completion) = 0;
+  /// \param[out] completion subrange of block containing the completion of partial
+  /// \param[out] rest subrange of block containing what completion does not cover
+  virtual Status Process(const std::shared_ptr<Buffer>& partial,
```

Comment:
Since this overload is doing something different than the other `Process` method, perhaps give it a different name (say `ProcessPartial`?).

Reply:
I'll rename it